# Data Analysis with Pandas: World Video Game Sales Data(cross-platform)
Mason Fryberger 01/19/2021

### Data Source and Analysis tools:
Data pulled from [Kaggle-Video-Game-Sales](https://www.kaggle.com/gregorut/videogamesales)

Data structured and analyzed with Pandas and NumPy
Dollar values in millions.

This Study attempts to answer the following questions:
- Which company is the most common video game publisher? (globally)
- What's the most common platform? (globally)
- What's the most common genre of game? (globally)
- What are the top 20 highest grossing games? (globally)
- For North Amercan (NA) game sales, what is the median?
    - what are the top ten games at or around the median?
- For the top selling game of all time (globally), what is the standard deviation of NA sales
  to the Global average
- The Nintendo Wii seems to have the most games published, how does its average sales compare
  to the sales of other platforms? 
  - overall?
  - individually?
  
- What Percent of the global marketshare is held by NA?
- What is the average rate of growth for the game publishing industry globally?
- Assuming a constant rate of growth and constant marketshare, what are the expected sales
  for NA in 5 years



Imports, Aliases, and Assignments


In [1]:
%matplotlib inline
import numpy as np 
import pandas as pd 
import matplotlib.pyplot as plt 

df = pd.read_csv('data-sets-csv/vgsales.csv')
pd.set_option('precision', 4)


In [2]:

df.set_index('Name')

Rank Platform    Year  \
Name                                                                       
Wii Sports                                            1      Wii  2006.0   
Super Mario Bros.                                     2      NES  1985.0   
Mario Kart Wii                                        3      Wii  2008.0   
Wii Sports Resort                                     4      Wii  2009.0   
Pokemon Red/Pokemon Blue                              5       GB  1996.0   
...                                                 ...      ...     ...   
Woody Woodpecker in Crazy Castle 5                16596      GBA  2002.0   
Men in Black II: Alien Escape                     16597       GC  2003.0   
SCORE International Baja 1000: The Official Game  16598      PS2  2008.0   
Know How 2                                        16599       DS  2010.0   
Spirits & Spells                                  16600      GBA  2003.0   

                                                         Genre   Publisher  \
Name                                                                         
Wii Sports                                              Sports    Nintendo   
Super Mario Bros.                                     Platform    Nintendo   
Mario Kart Wii                                          Racing    Nintendo   
Wii Sports Resort                                       Sports    Nintendo   
Pokemon Red/Pokemon Blue                          Role-Playing    Nintendo   
...                                                        ...         ...   
Woody Woodpecker in Crazy Castle 5                    Platform       Kemco   
Men in Black II: Alien Escape                          Shooter  Infogrames   
SCORE International Baja 1000: The Official Game        Racing  Activision   
Know How 2                                              Puzzle    7G//AMES   
Spirits & Spells                                      Platform     Wanadoo   

                                                  NA_Sales  EU_Sales  \
Name                                                                   
Wii Sports                                           41.49     29.02   
Super Mario Bros.                                    29.08      3.58   
Mario Kart Wii                                       15.85     12.88   
Wii Sports Resort                                    15.75     11.01   
Pokemon Red/Pokemon Blue                             11.27      8.89   
...                                                    ...       ...   
Woody Woodpecker in Crazy Castle 5                    0.01      0.00   
Men in Black II: Alien Escape                         0.01      0.00   
SCORE International Baja 1000: The Official Game      0.00      0.00   
Know How 2                                            0.00      0.01   
Spirits & Spells                                      0.01      0.00   

                                                  JP_Sales  Other_Sales  \
Name                                                                      
Wii Sports                                            3.77         8.46   
Super Mario Bros.                                     6.81         0.77   
Mario Kart Wii                                        3.79         3.31   
Wii Sports Resort                                     3.28         2.96   
Pokemon Red/Pokemon Blue                             10.22         1.00   
...                                                    ...          ...   
Woody Woodpecker in Crazy Castle 5                    0.00         0.00   
Men in Black II: Alien Escape                         0.00         0.00   
SCORE International Baja 1000: The Official Game      0.00         0.00   
Know How 2                                            0.00         0.00   
Spirits & Spells                                      0.00         0.00   

                                                  Global_Sales  
Name                                                            
Wii 

### Most common Game publisher

In [3]:
common_pub = df['Publisher'].mode()
common_pub


0    Electronic Arts
dtype: object

In [4]:
games_by_pub = df[df['Publisher'] == 'Electronic Arts'][['Name']].count()
games_by_pub

Name    1351
dtype: int64

We can see that Electronic Arts was the most common publisher over the time covered by the dataset with over 1,300 games published.

### Most Common Platform

We based our calclulation for the most common platform purely on the total number of titles published for the system in the years covered by the study. It is possible that a more accurate representation might be achieved using a coefficient of global sales in relation to other platforms however with out access to data regarding the Number of copies sold, this approach was disregarded and we have restricted our query to the data-set and number of occurences within it. 

In [5]:
common_plat = df['Platform'].mode()
common_plat

0    DS
dtype: object

In [6]:
games_on_ds = df[df['Platform'] == 'DS'][['Name']].count()
games_on_ds

Name    2163
dtype: int64

### Most Common Genre: (occurences in Dataset)

as with Most Common Platform we restrict ourselves to measureing occurences in the dataset, as calculations without access  to cost/unit and units-sold could not be considered accurate regarding presence/demand in the marketplace.


In [7]:
common_genre = df['Genre'].mode()
print(common_genre)
titles_in_genre = df[df['Genre'] == 'Action'][['Name']].count()
titles_in_genre


0    Action
dtype: object


Name    3316
dtype: int64

In conclusion the "Action" genre is the most common in the dataset with 3,316 occurences.


### Top 20 Highest Grossing games (globally)


In [ ]:
sorted_games_by_sales = df[['Name', 'Global_Sales']].set_index('Name').head(20)
print(sorted_games_by_sales)
sorted_games_by_sales.plot(kind="bar")


### Median NA Sales and top ten games at or near Median value.

For games in the top ten list with sales not equal to the median we sorted by equal number of titles above and below the median. 
on discovering that more than ten titles were at the median value we decided to plot the first ten returned by our query. values shown in millions.

In [ ]:
median_na_sales = df['NA_Sales'].median()
print(f'{median_na_sales} million')

ten_group_med = df[df['NA_Sales'] == median_na_sales][['Name', 'NA_Sales']].set_index('Name').head(10)
    
print(f'{ten_group_med}')

### Standard Deviation for Top Grossing game to Global Average

to determine how far above or below the Global average the top grossing game was we first Identified the game and global average and then used pandas built in std() method to get the standard deviation. by multiplying the difference between global_mean and NA_Sales by the standard deviation we get the number of deviations above or below as a (+/-)integer

In [ ]:
global_mean = df['Global_Sales'].mean()
print(f'{global_mean} Global Average in millions')

top_gross = df[df['Global_Sales'] == df['Global_Sales'].max()][['Name']].set_index('Name')
top_gross_val = df['Global_Sales'].max()
print(f'Top Grossing Game of all time: {top_gross.index[0]} with {top_gross_val} million in Global Sales')

standard_devi = df['Global_Sales'].std()
print(f'Standard Deviation: {standard_devi}')
top_gross_na_sales = df[df['Global_Sales'] == df['Global_Sales'].max()][['Name', 'NA_Sales']][['NA_Sales']].set_index('NA_Sales')
print(f'North American sales for Top Grossing game of all time: {top_gross_na_sales.index[0]}')

print(f'At {top_gross_na_sales.index[0]} {top_gross.index[0]} NA Sales are {(global_mean - top_gross_na_sales.index[0]) // standard_devi} standard deviations below the global average of {global_mean}')

### Nintendo Wii: average sales compared to other platforms.
while the assigned feature task claims that the Wii has most games published we know from our earlier analysis that it is actually the DS with most games published. 

here we have chosen to plot the global sales of all platforms in a barchart and then show the global sales for Nintendo Wii as a percentage of all global sales in a piechart.

In [ ]:
platforms = df.groupby('Platform')['Global_Sales'].sum().sort_values()
wii_sale = platforms[platforms == platforms['Wii']]
glb_sale = platforms[platforms != platforms['Wii']]

other_sales = pd.Series([glb_sale.sum()], index=["Other"])
wii_sale = wii_sale.append(other_sales)

fig, axs = plt.subplots(1,2)

print(f'Global Sales by Platform in millions: \n {platforms}')

platforms.plot(kind='bar', ax=axs[0]) 
wii_sale.plot(kind='pie', label="", ax=axs[1]) 


### What percent of Global marketshare is held by NA:
 
We can see this with a simple calculation by pulling the sum of NA sales and dividing it by the sum of global sales. 


In [ ]:
total_mkt = df['Global_Sales'].sum().round(3)
na_mkt = df['NA_Sales'].sum().round(3)
mkt_capture_na = na_mkt / total_mkt
print(f' North America holds {mkt_capture_na.round(2) * 100} % of the Global Market')

### Average Rate of Growth for Global Game Sales.



In [ ]:
by_year = df.groupby('Year')[['Global_Sales']].sum()
num_years = by_year['Global_Sales'].count()
first_sales = by_year['Global_Sales'][by_year.index[0]].round()
# print(f'first year of study Global_Sales in millions: {first_sales} ')
last_sales = by_year['Global_Sales'][by_year.index[-1]]
# print(f'last year of study Global_Sales in millions: {last_sales}')

growth = ((last_sales - first_sales ) / first_sales) * 100
print(f'Historical Growth of Global_Sales over {num_years} covered by the study:  {growth.round(3)} %')
# print(f'Global_Sales for {num_years} years covered by the study: in millions \n {by_year}')
